submodule.go

```diff
@@ -17,6 +18,9 @@ import (
 var (
 	ErrSubmoduleAlreadyInitialized = errors.New("submodule already initialized")
 	ErrSubmoduleNotInitialized     = errors.New("submodule not initialized")
+
+	// gitSubmoduleSSHRegex matches a submodule repository that connects over ssh
+	gitSubmoduleSSHRegex = regexp.MustCompile(`^\"?(ssh://)?[A-Za-z_]+@[a-zA-Z0-9._-]+:[a-zA-Z0-9./._-]+\/.*\"?$`)
```

Comment:
🤔  I have a few questions about this regex; 

* did you get this from a reference (if so can we list it in the comment)? 
* I feel like we should additionally be checking for `.git` at the ending
* Does this support the `git@{server}/...` format? (e.g. git@github.com:digitalocean/go-git.git)

Reply:
Looking at http://git-scm.com/book/en/v2/Git-on-the-Server-The-Protocols it feels like something closer to `(?:ssh|git@[-\w.]+):(\/\/)?(.*?)(\.git)` would get us what we want but I might be over simplifying the problem. Just wanted to throw this out as an idea though.

Reply:
I came up with the regex, what it was trying to accomplish was only matching on ssh server references. The regex you passed will work, but will also need to account for `\"` in the actual string (yes, I know, it's weird)

So more like:
https://regex101.com/r/zKXjEj/2

Reply:
I modifed and used your suggestion